File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefSerializableProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.convert.impl.SerializableConverter;
import gargoyle.ct.pref.CTPreferencesProvider;
import org.jetbrains.annotations.NotNull;

import java.io.Serializable;

public class CTPrefSerializableProperty<T extends Serializable> extends CTPrefProperty<T> {
    public CTPrefSerializableProperty(Class<T> type, @NotNull CTPreferencesProvider provider, String name) {
        this(type, provider, name, null);
    }

    public CTPrefSerializableProperty(Class<T> type, @NotNull CTPreferencesProvider provider, String name, T def) {
        super(type, new SerializableConverter<>(), provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefSerializableProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.convert.impl.SerializableConverter
import gargoyle.ct.pref.CTPreferencesProvider
import kotlin.reflect.KClass

class CTPrefSerializableProperty<T : Any> @JvmOverloads constructor(
    type: KClass<T>,
    provider: CTPreferencesProvider,
    name: String,
    def: T? = null
) : CTPrefProperty<T>(type, SerializableConverter(), provider, name, def)
